src/Http/Routing/src/ConfigureRouteHandlerOptions.cs

```diff
@@ -0,0 +1,28 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.AspNetCore.Routing
+{
+    internal class ConfigureRouteHandlerOptions : IConfigureOptions<RouteHandlerOptions>
+    {
+        private readonly IHostEnvironment _environment;
+
+        public ConfigureRouteHandlerOptions(IHostEnvironment environment)
+        {
+            _environment = environment;
+        }
+
+        public void Configure(RouteHandlerOptions options)
+        {
+            options.ThrowOnBadRequest = _environment.IsDevelopment();
```

Comment:
Should it be like this, in case someone calls configure before `AddRouting`?
```suggestion
            if (_environment.IsDevelopment())
            {
                options.ThrowOnBadRequest = true;
            }
```